mmdeploy/codebase/mmdet/models/roi_heads/single_level_roi_extractor.py

```diff
@@ -137,14 +137,22 @@ def single_roi_extractor__forward(ctx,
         device=target_lvls.device)
     target_lvls = torch.cat((_tmp, _tmp, target_lvls))
     for i in range(num_levels):
-        # use the roi align in torhcvision to accelerate the inference
-        # roi_align in MMCV is same as torchvision when pool mode is 'avg'
-        if backend == Backend.TORCHSCRIPT or self.roi_layers[
-                i].pool_mode == 'avg':
-            self.roi_layers[i].use_torchvision = True
         mask = target_lvls == i
         inds = mask.nonzero(as_tuple=False).squeeze(1)
-        roi_feats_t = self.roi_layers[i](feats[i], rois[inds])
+        rois_t = rois[inds]
+        # use the roi align in torhcvision
+        if backend == Backend.TORCHSCRIPT:
+            self.roi_layers[i].use_torchvision = True
+        if backend == Backend.ONNXRUNTIME:
+            ir_cfg = get_ir_config(ctx.cfg)
+            opset_version = ir_cfg.get('opset_version', 11)
+            offset = 0.5 if self.roi_layers[i].aligned is True else 0.0
+            if opset_version <= 16:
+                # use RoiAlign in onnxruntime and preprocess rois to
+                # make compatible with op_set 10
+                rois_t[:, 1:] = rois_t[:, 1:] - offset / self.roi_layers[
```

Comment:
Adding offset in symbolic function in `ops/roi_align.py` would be better. Since models which does not come from mmdetection might also need this fix.